### ISSUE TITLE
[BUG]: media_player not showing source and sound mode

### ISSUE THREAD

**Issue by Headkillah** [samuelspagl/ha_samsung_soundbar#11]
If i look in the soundbar integration i can see wich source and wich sound mode are active.

But if i click at the control elements on the external device (the media_player) it doesn´t show me the source / sound mode in the popup. Please look at the screenshot what i mean (outlined in red).

And further the calculation of the volume level in this view is not correct. (outlined in yellow)
![yassi_errors](https://github.com/samuelspagl/ha_samsung_soundbar/assets/24635716/eca90b0e-60d0-4d8e-aee5-a635a734d87b)

**Comment by samuelspagl** [comment]
I'll take a look at it this evening. 

How is the current volume level in this view not correct? :)

**Comment by Headkillah** [comment]
It's not correct because the level is at 6 ;) The volume sensor is showing correct value :)

**Comment by samuelspagl** [comment]
Did you set a max volume ?

**Comment by Headkillah** [comment]
It's not correct because the level is at 6 ;) The volume sensor is showing correct value :)

**Comment by samuelspagl** [comment]
Sorry the late response, for some reason I didn't get a notification.

So yeah that explains the issue.
What you're describing is basically a scaling issue:

The slider of the `media_player` is defined as a float between `0` and `1` and displayed from `0` to `100`. As the `maxVolume` is 30, this slider is scaled from 0 to 30. (1 being 30). Therefore the value of the slider is different than what the sensor is showing (the real value).

The only solution for that would be to set `maxVolume` to `100`.

**Comment by samuelspagl** [comment]
So I ran some tests on my end, and didn't have the same findings as you regarding those two. 

On my system it is displayed correctly. Right after starting the soundbar it takes some time until its mapped correctly, but like I mentioned for me it picked it up correctly. 

Has it ever displayed it correctly?

**Comment by Headkillah** [comment]
hmmm....

On my system the entities "input_preset" and "sound_mode_preset" are showing the correct values, but NOT the "media_player.name_of_soundbar" entity itself. It's not really bad but would be nice if it was displayed there :)

**Comment by samuelspagl** [comment]
Okay, at last I understood what you meant. The new commit in the PR should fix it. I checked it on my local instance and it seemed to work.

**Comment by samuelspagl** [comment]
@Headkillah do you want to check it out? :) 

One note to the "input source" part. If you are playing something with Spotify directly on the Soundbar, it will not list anything in the `input_source` as the API returns `null` there

**Comment by Headkillah** [comment]
> @Headkillah do you want to check it out? :) 
> 
> One note to the "input source" part. If you are playing something with Spotify directly on the Soundbar, it will not list anything in the `input_source` as the API returns `null` there

In my example the source is HDMI 1 (playing Disney+)  so it should be have to show the input source, right?

And the sound mode is adaptive sound, so I should see it there? :)

**Comment by Headkillah** [comment]
Looks good on screenshot :) Upload the changes and i will test it asap.

**Comment by samuelspagl** [comment]
@Headkillah the changes are already uploaded to the existing branch :)

**Comment by Headkillah** [comment]
> @Headkillah the changes are already uploaded to the existing branch :)

I'm working in a quality assurance department and test software, but maybe im dumb today. I've downloaded again the files from github, copied everything to the right places, changed the sensor thingy. BUT it looks like in my first screenshot. I see the right values only on the integration entry, look following screenshot.

![soundbar_scrsht](https://github.com/samuelspagl/ha_samsung_soundbar/assets/24635716/d599efd7-78a3-4cc1-bab3-c1857e845f49)

**Comment by samuelspagl** [comment]
@Headkillah Now I'm probably a bit dumb today. I don't know what the issue is in your image 😅 😂

**Comment by samuelspagl** [comment]
This is the change that should make it work:
[media_player.py](https://github.com/samuelspagl/ha_samsung_soundbar/blob/c61b2ba3f21330db30c5b81d3f33723fac6c1751/custom_components/samsung_soundbar/media_player.py#L181)

**Comment by Headkillah** [comment]
> @Headkillah Now I'm probably a bit dumb today. I don't know what the issue is in your image 😅 😂

I have to answer faster :D (Vielleicht sollte ich es auf deutsch erklären ;o))

When I go to your soundbar integration, I can see under Controls that everything is displayed correctly (see [screenshot 1 ](https://github.com/samuelspagl/ha_samsung_soundbar/issues/11#issuecomment-1893422649))

However, if I now call up the soundbar entity "media_player.barry" under Devices and Services, which is called "barry" in my case, then it is displayed as shown in this [screenshot](https://github.com/samuelspagl/ha_samsung_soundbar/issues/11#issue-2070169367)

**Comment by samuelspagl** [comment]
@Headkillah 
Maybe wäre das möglich :D,
But **true** developers only talk in english xD

Okay let's leave the jokes aside.
Could you go to `Developer Tools` and `States` and search for your `media_player._______` entity and show me what its attributes are while the device is turned on :)

**Comment by Headkillah** [comment]
@samuelspagl

Hope that information is good enough because I have only access through the companion app...

Or do you need more infos?

![Screenshot_20240118-100813_Home Assistant.jpg](https://github.com/samuelspagl/ha_samsung_soundbar/assets/24635716/d36a3ff3-ae97-4929-bc09-587820ea77d8)

**Comment by samuelspagl** [comment]
I need the whole list for this entity :) sorry. 

`sound_mode` and `source` should be there two times.

**Comment by Headkillah** [comment]
I forgot to mention something important: I have a Fire TV Stick in the HDMI 1 port in the soundbar, everything runs via this.

```
source_list:
  - digital
  - HDMI1
  - bluetooth
  - wifi
  - HDMI2
sound_mode_list:
  - standard
  - surround
  - game
  - adaptive sound
status: "on"
device_information:
  model: HW-Q995B
  manufacture: Samsung Electronics
  firmware_version: HW-Q995BWWB-1010.0
  device_id: xxxxxxxxxxxxxxxxxxxx
volume:
  level: 0.16666666666666666
  muted: false
woofer:
  level: 6
  connection: "on"
source:
  active_source: HDMI1
  supported_sources:
    - digital
    - HDMI1
    - bluetooth
    - wifi
    - HDMI2
sound_mode:
  active_sound_mode: adaptive sound
  supported_sound_modes:
    - standard
    - surround
    - game
    - adaptive sound
advanced_audio:
  night_mode: false
  bass_mode: true
  voice_amplifier: true
equalizer:
  active_preset: NONE
  supported_presets:
    - NONE
    - POP
    - JAZZ
    - CLASSIC
  action: setEQmode
  bands:
    - "0"
    - "0"
    - "0"
    - "0"
    - "0"
    - "0"
    - "0"
media:
  media_title: ""
  media_artist: ""
  media_cover_url: ""
  media_duration: 0
  media_position: 0
device_class: speaker
friendly_name: Barry
supported_features: 89485
volume_level: 0.16666666666666666
is_volume_muted: false
media_duration: 0
media_position: 0
media_title: ""
media_artist: ""
app_name: External Device
entity_picture: >-
  /api/media_player_proxy/media_player.barry?token=7006339a06bc508f0911dc935767cc156e9bd5a4ddd07ead99906edb3cb2545d&cache=e3b0c44298fc1c14
```

**Comment by samuelspagl** [comment]
Hmn interesting, it looks a bit different. 

Can you try to comment this part in the `media_player.py` 

```python
    @property
    def extra_state_attributes(self) -> Mapping[str, Any] | None:
        return {"device_info": self.device.retrieve_data}
```

Then restart your instance and take a look at it again :)

**Comment by Headkillah** [comment]
Yeah👍 now it looks right 🥳

![Screenshot_20240118-213038_Home Assistant.jpg](https://github.com/samuelspagl/ha_samsung_soundbar/assets/24635716/82e750a7-4d8f-48f1-84ee-7a57d91f6766)

**Comment by samuelspagl** [comment]
@Headkillah Interesting... 
It works for me even tho these lines are not commented. Which Ha version are you using? :)

**Comment by Headkillah** [comment]
I'm running HA on Proxmox version is 2024.1.3.

**Comment by samuelspagl** [comment]
Hi @Headkillah 
I hope everything works fine on your end. JFYI, I'm gonna release the fix officially this week.
If there's an issue still persisting just ping me :)

**Comment by samuelspagl** [comment]
Closed as #9 was merged.

**Comment by Headkillah** [comment]
Hi @samuelspagl 

Yes, sorry for late answer. Everything works like a charm now. Thanks for your hard work :)
